cli/flox/src/commands/init/mod.rs

```diff
@@ -809,4 +817,95 @@ mod tests {
             activate_mode: None,
         });
     }
+
+    /// format_customization() correctly converts InitCustomization to TOML
+    #[test]
+    fn test_format_customization() -> Result<()> {
+        // Create a test InitCustomization with various fields populated
+        let customization = InitCustomization {
+            hook_on_activate: Some("echo 'Activating environment'".to_string()),
+            profile_common: Some("export COMMON_VAR=value".to_string()),
+            profile_bash: Some("export BASH_VAR=value".to_string()),
+            profile_fish: Some("set -x FISH_VAR value".to_string()),
+            profile_tcsh: Some("setenv TCSH_VAR value".to_string()),
+            profile_zsh: Some("export ZSH_VAR=value".to_string()),
+            packages: Some(vec![CatalogPackage {
+                id: "test-package".to_string(),
+                pkg_path: "test.package".to_string(),
+                version: Some("1.0.0".to_string()),
+                systems: None,
+            }]),
+            activate_mode: None,
+        };
+
+        let toml_str = format_customization(&customization)?;
+
+        let parsed_toml = toml_str.parse::<toml_edit::DocumentMut>()?;
```

Comment:
nonblocking suggestion: I think it would be more readable to just `assert_eq!(toml_str, indoc! {r#"...multiline manifest...` rather than re-parsing the toml